### PR TITLE
adopters: add PatSnap

### PR DESCRIPTION
**Pull request by CaitinChen**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

add PatSnap

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by TomShawn — APPROVED** [review]
LGTM

File: adopters.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ This is a list of TiDB adopters in various industries.
 |[Maizuo](https://www.crunchbase.com/organization/maizhuo)|E-commerce||
 |[Mogujie](https://www.crunchbase.com/organization/mogujie)|E-commerce||
 |[Zhihu](https://en.wikipedia.org/wiki/Zhihu)|Knowledge Sharing|English [#1](https://pingcap.com/case-studies/lesson-learned-from-queries-over-1.3-trillion-rows-of-data-within-milliseconds-of-response-time-at-zhihu/), [#2](https://pingcap.com/case-studies/horizontally-scaling-hive-metastore-database-by-migrating-from-mysql-to-tidb); [Chinese](https://pingcap.com/cases-cn/user-case-zhihu/)|
+|[PatSnap](https://www.crunchbase.com/organization/patsnap)|Artificial Intelligence|English (https://pingcap.com/case-studies/why-we-chose-a-scale-out-data-warehouse-for-real-time-analytics); [Chinese](https://pingcap.com/cases-cn/user-case-zhihuiya/)|
```
Comment (TomShawn) [inline adopters.md:32]:
```suggestion
|[PatSnap](https://www.crunchbase.com/organization/patsnap)|Artificial Intelligence| [English](https://pingcap.com/case-studies/why-we-chose-a-scale-out-data-warehouse-for-real-time-analytics); [Chinese](https://pingcap.com/cases-cn/user-case-zhihuiya/)|
```

**Comment by CaitinChen** [comment]
@TomShawn Thanks.

**Pull request merged**